robolectric/src/main/java/org/robolectric/RobolectricTestRunner.java

```diff
@@ -521,6 +372,23 @@ ResModeStrategy getResModeStrategy() {
     return ResModeStrategy.getFromProperties();
   }
 
+  private void reportPerfStats(PerfStatsCollector perfStatsCollector) {
+    if (perfStatsReporters.isEmpty()) {
+      return;
+    }
+
+    Metadata metadata = perfStatsCollector.getMetadata();
+    Collection<Metric> metrics = perfStatsCollector.getMetrics();
+
+    for (PerfStatsReporter perfStatsReporter : perfStatsReporters) {
+      try {
+        perfStatsReporter.report(metadata, metrics);
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
   public static class HelperTestRunner extends SandboxTestRunner.HelperTestRunner {
```

Comment:
Our static analysis engine detects that your pull request includes a change that a random object is created and used only once. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-random-object-created-and-used-only-once-dmi-random-used-only-once), it is inefficient, and if you prefer unpredictable random numbers, `java.security.SecureRandom` is strongly recommended.